task4/VCS/Watcher.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace VCS
+{
+    /// <summary>
+    /// A class that monitors the file system
+    /// </summary>
+    public class Watcher
```

Comment:
If you have a field of Type that implements IDisposable - you should implement IDisposable too